### PR TITLE
Add support for AppInsights, interactive experience

### PR DESCRIPTION
**Pull request by DonJayamanne**
* Support App Insights
	* Had to refactor how connections are managed to support for more types of connections.

Adding AppId & AppKey in a single input seems clunky (delimited via comma).
But thats fewer prompts for the user.
I can change this to have 2 input prompts, one for AppId & next for AppKey.

addConnection

	* Once an App Insights connection has been added, I display the first few characters of the AppId
	Not displaying all of the AppId & Key, i think that's not something we want exposed.
	Not sure, what do you think?
	I personally think we should allow users to set up friendly display labels (like we can do today in Kusto Explorer)

Screen Shot 2021-04-06 at 02 41 59

* Prelimnary support for interactive experience
	* Support for *.kql or *.csl files (language features & executing queries)
	* Still some polishing required (hence not going to mention this just yet, but don't see any harm in putting it into the extension).

Screen Shot 2021-04-02 at 14 11 47
Screen Shot 2021-04-02 at 14 05 56

* Sunburst charts (I really need this)
sunburst

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,14 +37,16 @@
         "Visualization"
     ],
     "activationEvents": [
+        "onLanguage:kusto",
```
Comment (DonJayamanne) [inline package.json:40]: Now supporting language features & execution of queries from kql/csl files (not just notebooks)

File: package.json [hunk 2]
```diff
@@ -37,14 +37,16 @@
...
         "onLanguage:plaintext",
         "onView:kustoExplorer",
-        "onCommand:kusto.addCluster",
-        "onCommand:kusto.removeCluster",
+        "onCommand:kusto.executeSelectedQuery",
+        "onCommand:kusto.addConnection",
+        "onCommand:kusto.removeConnection",
         "onCommand:kusto.refreshNode",
-        "onCommand:kusto.changeNotebookDatabase"
+        "onCommand:kusto.changeDocumentConnection"
```
Comment (DonJayamanne) [inline package.json:49]: Refactored to add support for more types of connecetions, hence renamed to `Connection`.

File: src/client/visualization.tsx [hunk 3]
```diff
@@ -54,11 +54,28 @@ function getChartType(
     /**
     [1, "Visualization", "{"Visualization":"piechart","Title":null,"XColumn"…"]
     */
-    if (queryPropertiesTable._rows[0][1] !== 'Visualization') {
+    if (
+        queryPropertiesTable._rows[0][1] !== 'Visualization' &&
+        // This is how we get Visualization for AppInsights.
+        !(queryPropertiesTable._rows[0][0] as string).includes('Visualization')
```
Comment (DonJayamanne) [inline src/client/visualization.tsx:60]: The AppInsights endpoint doesn't return the same schema.

Comment (DonJayamanne) [reply src/client/visualization.tsx:60]: npm package doesn't seem to support AppInsights endpoints or I'm doing something wrong.

File: src/extension/kernel/kernel.ts [hunk 4]
```diff
@@ -26,8 +30,98 @@ export class Kernel implements NotebookKernel {
...
+    private static interactiveKernel?: Kernel;
+    public static get InteractiveKernel() {
+        return Kernel.interactiveKernel;
+    }
+    constructor(public readonly document: NotebookDocument) {
+        if (isKustoInteractive(document)) {
+            Kernel.interactiveKernel = this;
+        }
+    }
+    public async executeInteractiveSelection(textEditor: TextEditor): Promise<void> {
```
Comment (DonJayamanne) [inline src/extension/kernel/kernel.ts:42]: Lot of copy & paste (similar code in `executeCell`).
Will re-factor and ensure its using common code, for now i've left this hacky approach in here.
Requires polishing.

File: README.md [hunk 5]
```diff
@@ -13,16 +13,16 @@ A [Visual Studio Code](https://code.visualstudio.com/) [extension](https://marke
...
-* This extension augments Jupyter Notbooks with Kusto language features, when using the [Jupyter](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) extension.
-* The extension will automatically detect the cluster and database from cells containing the connection information `kql AzureDataExplorer://code;cluster='help';database='Samples'`.
-
 # Getting Started
 * Please install VS Code Insiders (stable is not yet supported)
 * Install this extension
 * Launch VS Code with the following command line `code-insiders --enable-proposed-api=donjayamanne.kusto`
-* Create a file with extension `*.knb`
-* Use the command `Configure Kusto Cluster/Database for Notebooks` to configure the Kusto connection.
+* Create a file with extension `*.knb` (or use the command `Create Kust Notebook`)
```
Comment (joyceerhl) [inline README.md:20]:
```suggestion
* Create a file with extension `*.knb` (or use the command `Create Kusto Notebook`)
```

File: README.md [hunk 6]
```diff
@@ -13,16 +13,16 @@ A [Visual Studio Code](https://code.visualstudio.com/) [extension](https://marke
...
+* Use the command `Configure Kusto Connection` to configure the Kusto connection.
+
+# Works with Jupyter Notebooks as well (when using [kqlmagic](https://pypi.org/project/Kqlmagic/))
+* This extension augments Jupyter Notbooks with Kusto language features, when using the [Jupyter](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) extension.
```
Comment (joyceerhl) [inline README.md:24]:
```suggestion
* This extension augments Jupyter Notebooks with Kusto language features, when using the [Jupyter](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) extension.
```

**Review by joyceerhl — COMMENTED** [review]
Two nits, will have more functional feedback tomorrow

**Pull request merged**